### PR TITLE
PP-2398 Refactored Transaction Detail View

### PR DESCRIPTION
**Pull request by maxcbc**
## WHAT
- Split transactionsCtrl.show into transactionDetailCtrl
- Split views into separate transaction_detail directory
- Switched to 3/4 1/4 column split
- Added amount to events table
- Switched displayed amounts on transaction detail and transaction list to use currency formatter (adds thousand separators)
- Removed refunded tick/cross from transaction details

### CODE REVIEW

File: app/utils/transaction_view.js [hunk 1]
```diff
@@ -71,14 +92,13 @@ module.exports = {
...
 
-    connectorData.eventStates = Object.keys(eventStates).map(function (str) {
-      var value = {}
-      value.text = changeCase.upperCaseFirst(str.toLowerCase())
-      if (str === filters.state) {
-        value.selected = true
-      }
-      return {'key': str, 'value': value}
-    })
+    connectorData.eventStates = []
```
Comment (ckalista) [inline app/utils/transaction_view.js]: Nice to see these little refactoring initiatives along the way.

File: app/views/transaction_detail/_details.html [hunk 2]
```diff
@@ -1,50 +1,58 @@
...
+
     {{#permissions.transactions_amount_read}}
     <tr>
-      <td>Amount:</td>
+      <td>Payment amount:</td>
       <td id="amount">{{amount}}</td>
     </tr>
     {{/permissions.transactions_amount_read}}
+
+    {{#refunded}}
```
Comment (ckalista) [inline app/views/transaction_detail/_details.html]: Have we confirmed that behaviour? Last time I spoke with @tillwirth it wasn't clear yet wether we wanted to hide the refund section of there was not refund on the charge.

File: app/utils/transaction_view.js [hunk 3]
```diff
@@ -121,17 +141,21 @@ module.exports = {
...
-      event.state_friendly = eventStates[event.state.status]
-      if (event.state_friendly) {
-        event.state_friendly = event.state_friendly.replace('AMOUNT', CURRENCY + (chargeData.amount / 100).toFixed(2))
-      }
       event.updated_friendly = dates.utcToDisplay(event.updated)
+      event.amount_friendly = event.amount ? asGBP(event.amount) : ''
+
+      const state = `${_.get(event, 'state.status')}`.toLowerCase()
+
+      if (!event.type || event.type.toLowerCase() === 'payment') {
```
Comment (ckalista) [inline app/utils/transaction_view.js]: I am not sure I understand the rational here - maybe if you could explain it tomorrow as I am sure I am missing something.

File: app/assets/sass/application.scss [hunk 4]
```diff
@@ -39,14 +39,13 @@ $grey-8: #fff;
 @import "modules/_transactions_filter";
 @import "modules/_dev_tokens";
 @import "modules/_transactions_list";
-@import "modules/_transaction_details";
+@import "modules/transaction_detail";
```
Comment (maxcbc) [inline app/assets/sass/application.scss]: error on my part, will correct it

File: app/views/transaction_detail/_details.html (hunk 2 above)
Comment (maxcbc) [inline app/views/transaction_detail/_details.html]: confirmed with till. We will go with this behaviour for now pending @soniaturcotte getting back

File: app/utils/transaction_view.js (hunk 1 above)
Comment (maxcbc) [inline app/utils/transaction_view.js]: One aims to please

File: app/controllers/transactions/transaction_detail_controller.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+'use strict'
```
Comment (maxcbc) [inline app/controllers/transactions/transaction_detail_controller.js]: :-D

**Pull request merged**